File: java/calendar/src/test/java/com/rbleggi/calendar/BookMeetingCommandTest.java
package com.rbleggi.calendar;

import org.junit.jupiter.api.Test;
import java.time.LocalDateTime;
import java.util.Set;
import static org.junit.jupiter.api.Assertions.*;

class BookMeetingCommandTest {
    @Test
    void executeBooksMeeting() {
        var calendar = new Calendar();
        var user = new User("1", "Alice");
        var meeting = new Meeting("m1", "Sync", LocalDateTime.of(2025, 1, 1, 9, 0), LocalDateTime.of(2025, 1, 1, 10, 0), Set.of(user));
        var command = new BookMeetingCommand(calendar, meeting);
        assertTrue(command.execute());
    }

    @Test
    void executeAddsMeetingToCalendar() {
        var calendar = new Calendar();
        var user = new User("1", "Alice");
        var meeting = new Meeting("m1", "Sync", LocalDateTime.of(2025, 1, 1, 9, 0), LocalDateTime.of(2025, 1, 1, 10, 0), Set.of(user));
        new BookMeetingCommand(calendar, meeting).execute();
        assertEquals(1, calendar.getAllMeetings().size());
    }
}

File: kotlin/calendar/src/test/kotlin/com/rbleggi/calendar/BookMeetingCommandTest.kt
package com.rbleggi.calendar

import java.time.LocalDateTime
import kotlin.test.Test
import kotlin.test.assertTrue
import kotlin.test.assertEquals

class BookMeetingCommandTest {
    @Test
    fun `execute books meeting`() {
        val calendar = Calendar()
        val user = User("1", "Alice")
        val meeting = Meeting(title = "Test", start = LocalDateTime.of(2025, 1, 1, 9, 0), end = LocalDateTime.of(2025, 1, 1, 10, 0), attendees = setOf(user))
        val command = BookMeetingCommand(calendar, meeting)
        assertTrue(command.execute())
    }

    @Test
    fun `undo removes booked meeting`() {
        val calendar = Calendar()
        val user = User("1", "Alice")
        val meeting = Meeting(title = "Test", start = LocalDateTime.of(2025, 1, 1, 9, 0), end = LocalDateTime.of(2025, 1, 1, 10, 0), attendees = setOf(user))
        val command = BookMeetingCommand(calendar, meeting)
        command.execute()
        command.undo()
        assertEquals(0, calendar.getAllMeetings().size)
    }
}
